Objects/descrobject.c

```diff
@@ -137,10 +131,13 @@ classmethod_get(PyMethodDescrObject *descr, PyObject *obj, PyObject *type)
 static PyObject *
 method_get(PyMethodDescrObject *descr, PyObject *obj, PyObject *type)
 {
-    PyObject *res;
-
-    if (descr_check((PyDescrObject *)descr, obj, &res))
-        return res;
+    if (obj == NULL) {
+        Py_INCREF(descr);
+        return (PyObject *)descr;
```

Comment:
Would you mind to use Py_NewRef()?

Reply:
I don't think we can, the bugfix needs to be backported to 3.9. Py_NewRef was added in 3.10. Unless you mean `Py_NewRef` for 3.11, 3.10 and manual backport for 3.9.